app/views/sessions/purchase_cart.html.erb

```diff
@@ -7,7 +7,11 @@
     </tr>
     <% if session[:cart] %>
         <% session[:cart].each do |item_id, quantity| %>
-        <% item = Item.find_by(id: item_id) %>
+            <% item = Item.find_by(id: item_id) %>
+            <!-- Line below generates receipt for purchasing this item, if possible -->
+            <% if logged_in? %>
```

Comment:
Primarily because a user doesn't need to be signed up in order to purchase items.

Someway, somehow, they do need to access their receipts.
Even if they were to sign up after the purchase, there would need to be a dependable way of gathering a receipt (which might be doable. If we, for example, saved the most recent receipt generated).

Ultimately, this 11 digit code seems the most straightforward. And as its space is 64 characters, the probability of anyone managing to get someone else's receipt is practically unbelievable.

This might be a problem if the userbase is primarily on their phones though. As I can imagine that might get clunky. On a laptop, I would ideally imagine them just copying and pasting.